src/protocol/kademlia/impl/kademlia_impl.cpp

```diff
@@ -434,7 +446,7 @@ namespace libp2p::protocol::kademlia {
     }
 
     peer_ids = peer_routing_table_->getNearestPeers(
-        NodeId(msg.key), config_.closerPeerCount * 2);
+        NodeId::hash(msg.key), config_.closerPeerCount * 2);
```

Comment:
```suggestion
        NodeId::hash(msg.key), std::max(config_.closerPeerCount * 2, config_.query_initial_peers));
```
Not a change request, only a suggestion if this could be considered as a reasonable edit

Reply:
If so, the change could be applied here and forth